### PR TITLE
CVE-2014-1745 and CVE-2015-3334

### PR DESCRIPTION
**Pull request by ada-phillips**
Researched and documented CVE-2014-1745 and CVE-2015-3334.

### CODE REVIEW

File: cves/CVE-2014-1745.yml [hunk 1]
```diff
@@ -54,8 +65,8 @@ fixes:
 - :commit: 9f88c15fbcd76321cfeedf3f4792534cb6041c4e
   :note: ''
 vccs:
-- :commit:
-  :note: 
+- :commit: d7df607030c7f32637e41a1ad00d3fd2bf94db1b
+  :note: Introduced as part of a performance enhancement
```
Comment (kbaumzie) [inline cves/CVE-2014-1745.yml]: Is this a valid claim? I cannot find the evidence that it was part of performance. Take a look at the introductory commit to understand more about why this was introduced.

File: cves/CVE-2014-1745.yml [hunk 2]
```diff
@@ -76,9 +87,11 @@ unit_tested:
 
     For the "fix" answer below, check if the fix for the vulnerability involves
     adding or improving an automated test to ensure this doesn't happen again.
-  answer: 
-  code: 
-  fix: 
+  answer: |
+    The existing unit tests did not cover this case explicitly, but more were 
+    added after the fix to ensure that the vulnerability does not return.
```
Comment (kbaumzie) [inline cves/CVE-2014-1745.yml]: looks like the only tests that were added were for around Mac OS specific. Do you feel this was properly aligned with this vulnerability? there is definitely no explicit testing around this bug

File: cves/CVE-2014-1745.yml (hunk 1 above)
Comment (ada-phillips) [reply cves/CVE-2014-1745.yml]: Discussion in the [original code review](https://codereview.chromium.org/82583005/#msg7) describes it as a "performance/clarity optimization".

File: cves/CVE-2014-1745.yml (hunk 2 above)
Comment (ada-phillips) [reply cves/CVE-2014-1745.yml]: The "discard-svg-font-face-crash.svg" and "discard-svg-font-face-crash-expected.svg" layout test files were added by the fix commit, and mimic the repro file that was included in the original bug report; they seem to pretty accurately address this vulnerability.

File: cves/CVE-2015-3334.yml [hunk 3]
```diff
@@ -76,9 +87,12 @@ unit_tested:
 
     For the "fix" answer below, check if the fix for the vulnerability involves
     adding or improving an automated test to ensure this doesn't happen again.
-  answer: 
-  code: 
-  fix: 
+  answer: |
+    Based on the code involved and the original bug report, it's clear that
+    unit tests were not involved in identifying or resolving this 
+    vulnerability. The issue was based entirely on presentation, not function.
```
Comment (mesternefeld) [inline cves/CVE-2015-3334.yml]: This is great, but were any tests added later to it?

File: cves/CVE-2015-3334.yml [hunk 4]
```diff
@@ -156,8 +174,10 @@ lessons:
...
   least_privilege:
-    applies: 
-    note: 
+    applies: true
+    note: |
+      The vulnerability dealt exclusively with the allocation of permissions,
+      and, more specifically, the reporting of this allocation to the user.
   frameworks_are_optional:
     applies: 
     note:
```
Comment (mesternefeld) [inline cves/CVE-2015-3334.yml]: Also, maybe this might be a secure by default as well as least privilege. What do you think?

Comment (ada-phillips) [reply cves/CVE-2015-3334.yml]: I don't think this one quite falls under secure by default. The two media permissions are not given by default, and giving one does not automatically give the other (although, that was a proposed solution to the issue, weirdly enough), the browser just doesn't display them to the user.

File: cves/CVE-2015-3334.yml [hunk 5]
```diff
@@ -52,10 +63,10 @@ fixes_vcc_instructions: |
...
 fixes:
 - :commit: 2b2a314fb6c3c2c1a947fa1686e79557503b97f4
-  :note: ''
+  :note: The vulnerability was resolved by fully separating the MIC and CAMERA permissions, so their values can be displayed separately.
 vccs:
-- :commit:
-  :note: 
+- :commit: 0bacc191a88137c1ba8602b681e96cf95ec825da
+  :note: The vulnerability was introduced when the MEDIA permission was (incompletely) split into separate MIC and CAMERA permissions.
 upvotes_instructions: |
```
Comment (Kosci) [inline cves/CVE-2015-3334.yml]: +3 upvotes

File: cves/CVE-2014-1745.yml [hunk 6]
```diff
@@ -54,8 +65,8 @@ fixes:
 - :commit: 9f88c15fbcd76321cfeedf3f4792534cb6041c4e
   :note: ''
 vccs:
-- :commit:
-  :note: 
+- :commit: d7df607030c7f32637e41a1ad00d3fd2bf94db1b
+  :note: Introduced as part of a performance enhancement
 upvotes_instructions: |
```
Comment (Kosci) [inline cves/CVE-2014-1745.yml]: +1 upvote

**Pull request merged**